src/test/groovy/unit/org/pih/warehouse/api/ProductApiTest.groovy

```diff
@@ -0,0 +1,178 @@
+import static org.hamcrest.Matchers.lessThanOrEqualTo
+
+class ProductApiTest extends Specification {
+
+    @Shared
+    String baseURI = System.getProperty("server.host")?:"http://localhost"
+
+    @Shared
+    String basePath = System.getProperty("server.path")?:"/openboxes"
+
+    @Shared
+    int port = System.getProperty("server.port")?Integer.valueOf(System.getProperty("server.port")):8080
+
+    @Shared
+    String username = System.getProperty("username")?:"admin"
+
+    @Shared
+    String password = System.getProperty("password")?:"password"
+
+    @Shared
+    String productId
+
+    @Shared
+    RequestSpecification requestSpecification
+
+    // Executed before the first spock test is executed
+    void setupSpec() {
+
+        RestAssured.basePath = basePath
+        RestAssured.baseURI = baseURI
+        RestAssured.port = port
+
+        JSONObject jsonObject = new JSONObject().
+                put("username", username).
+                put("password", password).
+                put("location", 1)
+
+        Response response = given().log().all().
+                contentType(ContentType.JSON).
+                body(jsonObject.toString()).
+                post("/api/login")
+
+        // Make sure we were successfully authenticated
+        response.then().assertThat().statusCode(200)
+
+        // We should have a cookie at this point
+        Cookie cookie = response.getDetailedCookie("JSESSIONID")
+
+        // FIXME For some reason ReqestSpecBuilder doesn't respect the config settings above
+        // so we need to set the baseURI, basePath, port again
+        requestSpecification = new RequestSpecBuilder().
+                setBaseUri(baseURI).
+                setBasePath(basePath).
+                setPort(port).
+                addCookie(cookie).
+                setAccept(ContentType.JSON).build()
+    }
+
+    void "should return a list of products"() {
+        given: "an authenticated user"
+        RequestSpecification request = given().spec(requestSpecification).log().all()
+
+        when: "we get products from the API"
+        // FIXME figure out better way to set up baseURI using RestAssured API
+        Response response = request.when().get("/api/products?max=10")
+
+        // FIXME Should be replaced by Product domain class once we
+        List<ProductResponse> products = response.then().log().all().extract().body().jsonPath().
+                getList("data", ProductResponse.class)
+
+        then: "expected a 200 response with at most 10 products"
+        response.then().assertThat().statusCode(200)
+        assertThat(products, is(notNullValue()))
+        assertThat(products.size(), is(lessThanOrEqualTo(10)))
+
+        // Get a single product to pass to the next test
+        when: "we attempt to share a product with the next test"
+        ProductResponse product = response.then().body().extract().jsonPath().getObject("data[0]", ProductResponse)
```

Comment:
Shouldn't we add new lines when we are calling multiple methods at once? I think we should try to avoid long lines like that (for example, on the frontend we have configured eslint with a maximum length of 100). It is harder to read for example in code review, when the line is too long.

Reply:
I don't mind breaking it up if there's a logical break (maybe after body and then again after jsonPath) but I also don't think it's necessary unless we go over the 100-120 character limit of the line. Which I'm guessing we are. 

But I don't have a strong feeling. In fact, I don't mind the one line per method (example below). I just don't want these tests to be 1000s of lines unnecessarily.
https://www.baeldung.com/rest-assured-response